pagopa/io-app
# fix: [IABT-1403,IOAPPCIT-9] Birth date is wrong in fiscal code component

## Short description
This PR fixes a bug on date formatting inside `FiscalCodeComponent`.
The date of birth is read from the `profile.date_of_birth`, if it's `undefined` then it's calculated from the fiscal code.

The `profile.date_of_birth` is saved on the profile as ISO 8601 string format at midnight (00:00:00.000):
`1977-05-22T00:00:00.000Z`

| Birth date 22/05/1977  | 
| ------------- | 
| <img src="https://user-images.githubusercontent.com/29163287/205049192-72fba6ab-78c4-41c0-ba30-d12eabefc030.png" width="200" />  | 

[Ref. IOAPPCIT-9 TimeZone](https://pagopa.atlassian.net/wiki/spaces/~634d02b3e14026a7397ce477/pages/585206146/IOAPPCITT-9+TimeZone)

## List of changes proposed in this pull request
To be timezone agnostic we crafted the birth date as dynamic string built only with year, month and date component of the UTC ISO string . We used two functions:
- `formatFiscalCodeBirthdayAsShortFormat`
- `formatFiscalCodeBirthdayAsAccessibilityReadableFormat`

<details>

<summary>More details</summary>

#### `formatFiscalCodeBirthdayAsShortFormat`
This function return `${day}/${month}/${year}` where `day`, `month` and `year` are UTC component of the date ISO string.

#### `formatFiscalCodeBirthdayAsAccessibilityReadableFormat`
This function return `${dayTranslationKey} ${date} ${monthTranslationKey} ${year}` where `date` and `year` are UTC component of the date ISO string while `monthTranslationKey` is the name of the month and  `dayTranslationKey` is the name of the day of the week. This function is specifically for the accessibility label. 

</details>

## how to test
- Use the dev-server by changing the [`date_of_birth`](https://github.com/pagopa/io-dev-api-server/blob/master/src/payloads/profile.ts#L51) and the [`fiscal_code`](https://github.com/pagopa/io-dev-api-server/blob/master/src/config.ts#L27)
- Check the birth date in FiscalCodeComponent (profile/tap the fiscal code at the top)
- Run `yarn test:tz`
